server/middleware/per-request.js

```diff
@@ -31,6 +31,16 @@ function perRequestContextFactory(options) {
   var scope = options.name || name;
   var enableHttpContext = options.enableHttpContext || false;
   var ns = LoopBackContext.createContext(scope);
+  // Monkey-patch LoopBackContext.getCurrentContext in order to fix issue #17
```

Comment:
Fixed, now the workaround is outside the middleware [factory], and it still works (and this time, this is what I expected) in my test app with 10 concurrent `curl` requests. And tests pass `only` with the workaround in place. They fail as soon as you comment the workaround out. 👍 
Also, I refactored the original method. I no longer monkey-patch it, which was not beautiful.
I bet it can be done without the proxy, but since I couldn't manage to do it, I'm fine with the proxy.